zerver/tornado/event_queue.py

```diff
@@ -769,24 +769,48 @@ def missedmessage_hook(
         internal_data = event.get("internal_data", {})
         sender_id = event["message"]["sender_id"]
 
+        # TODO/compatibility: Translation code for the rename of
+        # `wildcard_mention_push_notify` to `stream_wildcard_mention_push_notify`.
+        # Remove this when one can no longer directly upgrade from 7.x to main.
+        stream_wildcard_mention_push_notify = False
+        if "stream_wildcard_mention_push_notify" in internal_data:
+            stream_wildcard_mention_push_notify = internal_data.get(
+                "stream_wildcard_mention_push_notify"
+            )
+        elif "wildcard_mention_push_notify" in internal_data:
+            stream_wildcard_mention_push_notify = internal_data.get("wildcard_mention_push_notify")
+
+        # TODO/compatibility: Translation code for the rename of
+        # `wildcard_mention_email_notify` to `stream_wildcard_mention_email_notify`.
+        # Remove this when one can no longer directly upgrade from 7.x to main.
+        stream_wildcard_mention_email_notify = False
+        if "stream_wildcard_mention_email_notify" in internal_data:
+            stream_wildcard_mention_email_notify = internal_data.get(
+                "stream_wildcard_mention_email_notify"
+            )
+        elif "wildcard_mention_email_notify" in internal_data:
+            stream_wildcard_mention_email_notify = internal_data.get(
+                "wildcard_mention_email_notify"
+            )
```

Comment:
Or, more concisely,
```py
# TODO/compatibility: `stream_wildcard_mention_email_notify` was called
# `wildcard_mention_email_notify` earlier. Remove this when ...
stream_wildcard_mention_email_notify = internal_data.get(
        "stream_wildcard_mention_email_notify", False
    ) or internal_data.get(
        "wildcard_mention_email_notify", False
    )
```

Reply:
I think we should keep it consistent with other existing `TODO/compatibility` codeblocks in this file. Or change all of them in a prep commit ?

To me, the current codeblock looks more readable.

Reply:
Hmm right. I was mostly thinking about how to avoid long code, since we need to put this compatibility code in multiple places. But the current version is OK as well.